servers/services/src/main/java/org/projectnessie/services/rest/TreeResourceWithAuthorizationChecks.java

```diff
@@ -135,8 +133,12 @@ protected void checkHashOnRef(String refName, String hash) throws NessieNotFound
         // Special case, mostly for tests:
         // When the default branch is deleted, we cannot verify whether the branch already exists
         // (it does not, so the check would fail).
-        if (refName.equals(getConfig().getDefaultBranch()) && hash == null
-            || hash.equals(getStore().noAncestorHash().asString())) {
```

Comment:
The `equals` should have probably be the other way around (as `hash` might be `null`??)

Reply:
yep, `hash` ends up being null